src/Search.php

```diff
@@ -4620,6 +4620,22 @@ public static function addWhere($link, $nott, $itemtype, $ID, $searchtype, $val,
                     break;
             }
         }
+
+        // In some cases, we will need to replace the WHERE condition by a subquery.
+        // This is needed to avoid false positif for  for negative conditions on
+        // children items (e.g. search for tickets that DON'T have a given group)
+        // See https://github.com/glpi-project/glpi/pull/13684 for detailed examples
+        $use_where_subquery = false;
```

Comment:
Maybe a more precise name (I tried to keep is as short as possible)
```suggestion
        $use_subquery_on_id_search = false;
```